v2/src/lib.rs

```diff
@@ -832,51 +790,54 @@ fn key_exprs_to_influx_regex(path_exprs: &[&keyexpr]) -> String {
     result
 }
 
-fn time_from_parameters(t: &str) -> ZResult<Option<(f64, f64)>> {
+fn timerange_from_parameters(p: &str) -> ZResult<Option<String>> {
     use zenoh::selector::{TimeBound, TimeRange};
-    let time_range = t.time_range()?;
-
-    let mut start_time: f64 = 0_f64;
-    let mut stop_time: f64 = Utc::now().naive_utc().timestamp() as f64;
-
+    let time_range = p.time_range()?;
+    let mut result = String::new();
     match time_range {
         Some(TimeRange(start, stop)) => {
             match start {
                 TimeBound::Inclusive(t) => {
-                    start_time = calculate_time(t)?;
+                    result.push_str("start:");
+                    write_timeexpr(&mut result, t, 0);
                 }
                 TimeBound::Exclusive(t) => {
-                    start_time = calculate_time(t)? + 1_f64;
+                    result.push_str("start:");
+                    write_timeexpr(&mut result, t, 1);
                 }
                 TimeBound::Unbounded => {}
             }
             match stop {
                 TimeBound::Inclusive(t) => {
-                    stop_time = calculate_time(t)? + 1_f64;
+                    result.push_str(", stop:");
+                    write_timeexpr(&mut result, t, 1);
                 }
                 TimeBound::Exclusive(t) => {
-                    stop_time = calculate_time(t)?;
+                    result.push_str(", stop:");
+                    write_timeexpr(&mut result, t, 0);
                 }
                 TimeBound::Unbounded => {}
             }
         }
-        None => {
-            return Ok(None);
-        }
+        None => return Ok(None),
     }
-    Ok(Some((start_time, stop_time)))
+
+    Ok(Some(result))
 }
 
-fn calculate_time(tx: TimeExpr) -> ZResult<f64> {
-    let now_time = Utc::now().naive_utc().timestamp() as f64;
-    match tx {
+fn write_timeexpr(s: &mut String, t: TimeExpr, i: u64) {
+    use humantime::format_rfc3339_nanos;
+    use std::fmt::Write;
+    match t {
         TimeExpr::Fixed(t) => {
-            let time_in_subsecs = t
-                .duration_since(UNIX_EPOCH)
-                .expect("Time went backwards")
-                .as_secs_f64();
-            Ok(time_in_subsecs)
+            let tm = t + Duration::from_millis(i * 10); //adding 10ms for timebound
```

Comment:
I think 10ms is already a too big gap.
What we want here is a distinction between inclusive or exclusive timestamp, for the case a timestamp in the database has the exact same value than a stop/start range value. Knowing that in Flux language, the `range.start` is always inclusive and the `range.stop` is always exclusive.

So I think 1ns is the most accurate distinction we can do to include or exclude a timestamp.
I.e. the `i: u64` parameter could be replaced with a `add_one_ns: bool` param

Reply:
Added the changes.